src/Language/PureScript/TypeChecker/TypeSearch.hs

```diff
@@ -76,6 +77,30 @@ checkSubsume unsolved env st userT envT = checkInEnvironment env st $ do
   -- Finally, check any constraints which were found during elaboration
   Entailment.replaceTypeClassDictionaries (isJust unsolved) expP
 
+accessorSearch
```

Comment:
I'd like to wait for a third occurrence before drying it up. For the instance check I don't think it comes up when applying the subsumption rule to `{| u1} -> u2`. 

It would definitely be necessary if we do the thing I outlined in the comment below and attempt to unify the accessor with the expected return type for the typed hole.